dask/dataframe/io/sql.py

```diff
@@ -64,7 +68,9 @@ def read_sql_table(table, uri, npartitions, index_col, limits=None, columns=None
     if not isinstance(index_col, six.string_types + (elements.Label,)):
         raise ValueError('Use label when passing an SQLAlchemy instance'
                          ' as the index (%s)' % index)
-    if isinstance(npartitions, int):
+    if divisions is None and npartitions is None:
+        raise TypeError('Must supply either divisions or npartitions')
```

Comment:
Maybe try an exclusive or here?  Not sure what the right way to do this is.  Maybe the following:

    if (divisions is None) + (npartitions is None) != 1:
        ...